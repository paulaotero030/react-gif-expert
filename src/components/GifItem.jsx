import PropTypes from 'prop-types';

export const GifItem = ({ title, url, onClick }) => {
  return (
    <div className='card' onClick={onClick}>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
