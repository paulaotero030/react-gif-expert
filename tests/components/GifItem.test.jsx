import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {
  const title = 'Disney Princess Beauty GIF by Disney';
  const url =
    'https://media1.giphy.com/media/RLVHPJJv7jY1q/giphy.gif?cid=3651eed2e5iar3hh2engbqqg06v5obra3wi81u227k2oxbud&ep=v1_gifs_search&rid=giphy.gif&ct=g';

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });
  test('debe de mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
