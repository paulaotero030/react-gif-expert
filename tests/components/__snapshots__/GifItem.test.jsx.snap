// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Disney Princess Beauty GIF by Disney"
      src="https://media1.giphy.com/media/RLVHPJJv7jY1q/giphy.gif?cid=3651eed2e5iar3hh2engbqqg06v5obra3wi81u227k2oxbud&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Disney Princess Beauty GIF by Disney
    </p>
  </div>
</div>
`;
